app/kumactl/cmd/install/render_helm_files.go

```diff
@@ -119,12 +119,24 @@ func generateOverrideValues(args interface{}) map[string]interface{} {
 			}
 			root = root[n].(map[string]interface{})
 		}
-		root[splitTag[tagCount-1]] = value
+		root[splitTag[tagCount-1]] = adjustType(value)
 	}
 
 	return overrideValues
 }
 
+// If the parameter value is map it has to be of a type map[string]interface{} therefore we need to convert it
+func adjustType(value interface{}) interface{} {
+	if m, ok := value.(map[string]string); ok {
```

Comment:
Hmm, can it be `map[string]int`? then we need another conversion like this? Is there a way togeneralise this using  introspection. Like if the type is not `interface` then do the conversion?

Reply:
No it can't, at least for now there is no such type in `InstallControlPlaneArgs`.
Due to the lack of generics in Go, I see no better way to generalize this.

```
Like if the type is not interface then do the conversion?
```
everything can be cast to `interface{}`